### ISSUE TITLE
Consolidate usage of "Target Group" vs "TargetList"

### ISSUE THREAD

**Issue by jchate6** [TOMToolkit/tom_base#679]
As far as I can tell we use the terms "Target List" and "Target Group" more or less interchangeably with "Group" tending to be more user facing than "list". This inconsistency can be confusing and add unnecessary complexity for TOM users and especially TOM admin. 

I'd recommend a refactor in which we choose a single term for this idea and use consistent variable names throughout the toolkit and documentation.

(I'd also like to see a similar refactor for alias vs target_name").

**Comment by rachel3834** [comment]
I agree, and I'd favor the term targetlist rather than group, as in my
experience this is the more common parlance among astros.

**Comment by jchate6** [comment]
My only counter arguments in favor of "Group" over "List" are that 

1. Group is currently the more user facing term while target list is mostly in the code.
2. A traditional target "list" as most astronomers would use the term is possibly only a subset of the use cases for target "Groups" as they exist in TOM Toolkit. Referring to these as "target lists" might lead to a very limited interpretation of what these are. In Tom Toolkit, a group is much more akin to a high level "tag" than what I would consider a traditional target list with each target being a member of many groups.

**Comment by rachel3834** [comment]
Those are good points, especially point 2.  I do worry that currently users
have shown a little confusion over what the target groups are for at the
moment, for essentially the opposite reason: they were looking to create a
target list and didn't think about groups as a way to do this.  But perhaps
that's something we can overcome in the documentation - likely different
people would favor different options.  We could do a poll through the TOM
Slack channel and see what they vote for.

**Comment by jchate6** [comment]
The complicating factor for all of this is the existence of `UserGroups` which are also called "groups" and `targetgroupings` in the URLs

**Comment by jchate6** [comment]
Eliminate any use of "group" or "list" by themselves in the code.
These should be replaces with "ObjectGroup" where "object" is the model being grouped:
`TargetGroup`
`UserGroup`
`ObservtionGroup`
etc.
